### DISCUSSION TITLE
Should not NotEmailPasswordPolicyProvider be case-insensitive?

### DISCUSSION THREAD

**Discussion by osorionicolas** [keycloak/keycloak#17280] [Q&A]
Hi everyone,

Yesterday I noticed that the "Not Email" password policy is not validating case and I have the doubt if it should. As I understand emails could be only lowercase in Keycloak.

e.g. test@test.com != Test@test.com, so it's a valid password.

Is this behaviour ok?

Thanks!

**Comment by zinki97** [comment]
I support the proposal and extend it to "notUsername".

We have the password policy "at least 1 uppercase letter" - email addresses and usernames are stored lowercase.

=> "notUsername" and "notEmail" are both useless in combination with "at least 1 uppercase letter", because it can never occur.
